---
title: "RNA Sequencing in Pulmonary Arterial Hypertension"
subtitle: "BMIN503/EPID600 Final Project"
author: "Amber Meservey"
format: html
editor: visual
number-sections: true
embed-resources: true
---

------------------------------------------------------------------------

Use this template to complete your project throughout the course. Your Final Project presentation will be based on the contents of this document. Replace the title/name above and text below with your own, but keep the headers. Feel free to change the theme and other display settings, although this is not required. I added a new sentence

## Overview {#sec-overview}

Give a brief a description of your project and its goal(s), what data you are using to complete it, and what two faculty/staff in different fields you have spoken to about your project with a brief summary of what you learned from each person. Include a link to your final project GitHub repository.

```{r}
## Pulmonary arterial hypertension (PAH). is a disease characterized by pathologically increased pressures in the pulmonary vascular bed. Despite recent treatment advances, the disease remains associated with considerable morbidity and mortality. Currently disease diagnosis and prognosis relies simultaneously on invasive hemodynamic measurements and crude clinical associations. Accordingly, there is considerable interest in developing biomarkers for use in diagnosing, distinguishing subtype, treating and following treatment response in (PAH). Our group has previously looked at plasma protein biomarkers in a small clinical trial population of 84 participants with pulmonary arterial hypertension. The next phase of our  project expands on this prior work by examining RNA sequencing in whole blood samples collected from participants at 2 time points during the trial. Although I will ultimately be assessing RNA sequencing data from our own cohort, I will use this project as an opportunity to work through an RNA sequencing data set from a publically available dataset from patients with PAH (as our RNA raw data is not yet publically available). Because our RNA sequencing is from peripheral whole blood, I will work through a publically available dataset obtained from a different tissue set (either pulmonary artery smooth muscle cells or lung tissue) to then later compare whether our findings are similar in peripheral blood markers. 

## The two faculty/staff I've met with regarding this project are Dr. Steven Kawut in Pulmonary Hypertension and Dr. Rui Feng in Biostatistics. Dr. Kawut would be considered my content expert, as he is a master clinician-researcher in PAH, as well as the lead on this clinical trial. He helps to direct appropriate clinical questions being asked of this data. Dr Rui Feng has expertise in RNA sequencing methods as well as statistical methods informing their use. She has been helpful in helping me understand how to format and use appropriate packages for this data.


```

```{r}
## https://github.com/ames78/BMIN503_Final_Project/blob/Introduction/final_project_Meservey.qmd
```

## Introduction {#sec-introduction}

Describe the problem addressed, its significance, and some background to motivate the problem. This should extend what is in the @sec-overview.

Explain why your problem is interdisciplinary, what fields can contribute to its understanding, and incorporate background related to what you learned from meeting with faculty/staff.

```{r}
## Pulmonary arterial hypertension (PAH) is a disease of pathologically elevated pressures in the pulmonary arterial bed, leading to right heart failure and ~20% mortality at 3 years. Despite availability of several pharmacologic classes of medications for PAH, treatment algorithms remain largely etiology-agnostic and anchored on disease severity as opposed to disease entity. Tools to help discriminate disease etiology and predict treatment response are needed in order to better tailor treatment algorithms and to improve patient outcomes. We anticipate that the investigation of blood-based biomarkers represents an important means to achieving this goal. Our initial step toward exploring candidate biomarkers in PAH will be a retrospective examination of the baseline and change in expression of a subset of candidate protein biomarkers in an available small clinical trial of patients with PAH.  Our findings will help direct future larger and more costly studies to test and if appropriate validate potential candidate protein biomarkers.

## Following diagnosis of PAH by right heart catheterization (RHC), it is commonly classified by specific etiology, predominately idiopathic PAH (IPAH, ~40% of cases), followed by connective tissue disease-associated PAH (CTD-PAH, ~20% of cases) and PAH associated with several less common systemic diseases. Although all forms of PAH result from aberrant vascular remodeling and vasoconstriction in the pulmonary arteries, the underlying mechanisms are incompletely understood and likely mediated by each specific disease etiology. With a 4:1 predominance in female patients and increasing evidence for a role of adipose tissue in PAH, hormonal and metabolic pathways are also thought to play significant roles in PAH pathophysiology. This complex and heterogenous mechanistic milieu alongside significant mortality, reliance on invasive means of monitoring disease and an untailored treatment armamentarium makes PAH a disease for which investigation of non-invasive biomarkers is both justifiable and urgent. 

## Thus, I will assess RNA sequencing patterns by several important pathologic and phenotypic characteristics: RNA expression by PAH etiology, by sex, by BMI (if available), by severity metrics (such as six minute walk or hemodynamic measurements, if available) and by medication class (if available). Though I plan to look at these in my own cohort, for the purposes of this project I will look in a publically available database in another tissue such as lung or pulmonary artery smooth endothelial cells, so as to ascertain whether similar patterns are seen in both tissue types. 

```

## Methods {#sec-methods}

Describe the data used and general methodological approach used to address the problem described in the @sec-introduction. Subsequently, incorporate full R code necessary to retrieve and clean data, and perform analysis. Be sure to include a description of code so that others (including your future self) can understand what you are doing and why.

```{r}
## First- gain access to raw or normalized RNA sequencing counts from a publically available GEO dataset. Options I found that would be of interest include one that contains info available from pulmonary smooth artery cells and another of lung tissue. Then I will need to organize the colData and countData appropriately to begin running RNA seq packages for differential expression by sex, PAH etiology, etc. 
```

```{r}
### Download GEO Query in Bioconductor
if (! requireNamespace("BiocManager", quietly = TRUE))
    install.packages("BiocManager")
BiocManager::install("GEOquery")
```

```{r}
## Load GEO Query
library(GEOquery)
```

```{r}
## Select dataset of interest: PAH dataset, RNA-seq collected from pulmonary smooth artery cells
## ALTERNATIVE: GSE254617
## Download GEO Data: 
data <- getGEO("GSE243193", GSEMatrix = TRUE)
data <- data[[1]]

## View the available phenotypic data:
exprs(data)
sample_metadata <- pData(data)
head(sample_metadata)

## View the expression data as a dataframe
expression_data <- exprs(data)
expression_dataframe <- as.data.frame(expression_data)
```

```{r}

data1 <- getGEO("GSE254617", GSEMatrix = TRUE)
data1 <- data1[[1]]
## View the available phenotypic data:
exprs(data1)
sample_metadata1 <- pData(data1)
head(sample_metadata1)

## View the expression data as a dataframe
expression_data1 <- exprs(data1)
expression_dataframe1 <- as.data.frame(expression_data1)

## Alternative to get raw data: Download data in CSV 
file_path <- "/Users/ambermes/Desktop/GSE254617_txiCounts.csv.gz"
expression_data <- read.csv(gzfile(file_path), header = TRUE, row.names = 1)

file_path_smooth <-"/Users/ambermes/Desktop/GSE243193_PAEC_RNAseq_Counts_GEO2.09.11.23.csv.gz"
expression_data_smooth<- read.csv(gzfile(file_path_smooth), header = TRUE, row.names = 1)

```

\

## Results {#sec-results}

Describe your results and include relevant tables, plots, and code/comments used to obtain them. You may refer to the @sec-methods as needed. End with a brief conclusion of your findings related to the question you set out to address. You can include references if you'd like, but this is not required.

## Conclusion

This the conclusion. The @sec-results can be invoked here.
